src/transformers/modeling_tf_utils.py

```diff
@@ -78,6 +78,8 @@
 if TYPE_CHECKING:
     from . import PreTrainedTokenizerBase
 
+os.environ["TF_USE_LEGACY_KERAS"] = "1"  # Compatibility fix to make sure tf.keras stays at Keras 2
```

Comment:
We shouldn't just override an environment variable like this - this can lead to all sorts of unexpected behaviours on the users end e.g. if they explicitly set it to "0" in their own environment before running a script. 

Instead, we should:
* Check if it's already set
* If it's set to 1, then no worries
* If it's set to 0, raise an error saying this isn't compatible with transformers
* If it's unset, set it to 1

Reply:
Done! I raise a warning instead of an error because I expect it to be relatively uncommon for people to explicitly set the env var to 0 (as 0 just maintains the default behaviour). Also, it might be possible to use some objects like tokenizers from `transformers` while still using Keras 3, so I don't want to forcibly raise an error every time `modeling_tf_utils` is touched.